vyper/codegen/core.py

```diff
@@ -923,6 +923,11 @@ def is_return_from_function(node):
         return True
     if isinstance(node, (vy_ast.Return, vy_ast.Raise)):
         return True
+    if isinstance(node, vy_ast.If):
+        return_if_body = any(is_return_from_function(n) for n in node.body)
+        return_else_body = any(is_return_from_function(n) for n in node.orelse)
+        if return_if_body and return_else_body:
```

Comment:
Think this needs to be "if statement has a return, or it has an else statement and that also has a return" because an if statement might not have an else statement

Reply:
It implicitly already checks "it has an else statement".

If `node.orelse` is an empty list (in other words, if statement doesn't have else statement), `return_else_body` will be `False`.

```python
>>> any([])
False
>>> any(bool(n) for n in [])
False
```

Reply:
> Needs some tests for just an if statement with no else statement, and also an if statement with elif and then an else statement

There's already a test for if statement with no else statement:

```python
    """
@external
def foo() -> int128:
    if False:
        return 123
    return 333
    """,
```

But no test for `elif` case yet. I'll add it to the PR soon.

Reply:
Okay great, thanks for looking

Reply:
note the commentary in the [python ast documentation](https://docs.python.org/3/library/ast.html#ast.If)

> elif clauses don’t have a special representation in the AST, but rather appear as extra [If](https://docs.python.org/3/library/ast.html#ast.If) nodes within the orelse section of the previous one.

so `orelse` is always either an empty list or a list with one ("if") node in it.